Debug console is not working well after last release

Type: <b>Bug</b>

I start debugging, and I can use the debug console as normal, but when I restart debugging, the console stops working. I have to stop, and debug again.
However, the behaviour is unstable, and at times the debug bar does not respond, so I have to restart VSCode.

When I work without extensions, the behaviour is similar.

OS: Ubuntu 22.04.5 LTS

VS Code version: Code 1.97.0 (33fc5a94a3f99ebe7087e8fe79fbe1d37a251016, 2025-02-04T22:41:26.688Z)
OS version: Linux x64 6.8.0-52-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5600H with Radeon Graphics (12 x 4164)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off<br>webnn: disabled_off|
|Load (avg)|1, 2, 1|
|Memory (System)|15.03GB (7.95GB free)|
|Process Argv|--crash-reporter-id d0a222cb-a408-4f0e-a286-bea090c7e7aa|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.5.0
vscode-markdownlint|Dav|0.58.2
gitlens|eam|16.2.2
copilot|Git|1.265.0
copilot-chat|Git|0.24.0
todo-tree|Gru|0.0.226
wordcounter|kir|2.4.4
git-graph|mhu|1.30.0
debugpy|ms-|2025.0.0
python|ms-|2025.0.0
vscode-pylance|ms-|2025.2.1
jupyter|ms-|2025.1.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.1.0
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
vscode-icons|vsc|12.11.0
markdown-pdf|yza|1.5.0
dictionary-completion|yzh|1.3.1
markdown-all-in-one|yzh|3.6.2


</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
cppperfnew:31000557
dwnewjupytercf:31046870
nativerepl1:31139838
pythonrstrctxt:31112756
nativeloc2:31192216
iacca1:31171482
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530
6074i472:31201624
customenabled:31232589
9064b325:31222308
copilot_t_ci:31222730

```

</details>

<!-- generated by issue reporter -->

> For me (VScode version 1.97.0), this problem occurs when I am debugging python code. When I first run the debugger the debug console accepts inputs, but if I use the Restart button to restart the debugger (or Ctrl+Shift+F5), the debug console will become inactive. However, if I Stop the debugger, and then start it again, the debug console will be active again. Inactive means that I cannot write anything into the text field, but using the arrows will bring up recent lines, which can be executed.
> 
> Downgrading to 1.96.4 (I am on Ubuntu 24.10: sudo apt install code=1.96.4-1736991114) resolves the issue.
> 
> The output of using **Developer: Toggle Developer Tools** with 1.97.0:
> 
> at line lifecycle.ts:453 Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked! at Nfi.add (lifecycle.ts:453:18) at variablesView.ts:589:27
> 
> ![Image](https://github.com/user-attachments/assets/f53ac6a7-bde1-43b5-b5db-abf3a02d78cf)
> 
> Additionally using version 1.96.4 (the one which works) has more errors shown in the output:
> 
> at line lifecycle.ts:453 Error: Trying to add a disposable to a DisposableStore that has already been disposed of. The added object will be leaked! at Nfi.add (lifecycle.ts:453:18) at variablesView.ts:589:27
> 
> at loq.ts:445 ERR Invalid debug adapter: Error: Invalid debug adapter at vVt.p (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:748:95) at vVt.$acceptDAExit (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:748:35) at Odt.S (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:3144:41680) at Odt.Q (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:3144:41458) at Odt.M (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:3144:40546) at Odt.L (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:3144:39783) at oee.value (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:3144:38444) at x.B (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:30:748) at x.fire (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:30:967) at K6.fire (vscode-file://vscode-app/usr/share/code/resources/app/out/vs/workbench/workbench.desktop.main.js:762:10348)
> 
> at lifecycle.ts:762 Error: Trying to add a disposable to a DisposableMap that has already been disposed of. The added object will be leaked! at ts.set (lifecycle.ts:762:17) at debugSession.ts:1621:19 at Array.map () at BIs.run (debugSession.ts:1615:31)
> 
> ![Image](https://github.com/user-attachments/assets/08129b27-1804-4aa1-a3ad-ae73b8b3d91c)

thanks so much for this 😸. Did notice it happens when I restart the debugging session

> I am debugging Python.
> 
> ![Image](https://github.com/user-attachments/assets/896c4af0-7cf5-45c6-8507-b06d50e6bc13) [ ![Image](https://github.com/user-attachments/assets/896c4af0-7cf5-45c6-8507-b06d50e6bc13) ](https://private-user-images.githubusercontent.com/83649022/411459687-896c4af0-7cf5-45c6-8507-b06d50e6bc13.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyMTczODcsIm5iZiI6MTczOTIxNzA4NywicGF0aCI6Ii84MzY0OTAyMi80MTE0NTk2ODctODk2YzRhZjAtN2NmNS00NWM2LTg1MDctYjA2ZDUwZTZiYzEzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEwVDE5NTEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyN2Q2ZDEwN2QxZDZmMjM2MzViNTQ2OWE0OWM3NDIxNmU5NDRjNjNiYTQ3MTJiZDdjN2RmNTM3ODhiOWI0MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6LSk9VvdWDmEebDV2H-jEzJy32-oY1x3lPXQWO9ByBc) [ ](https://private-user-images.githubusercontent.com/83649022/411459687-896c4af0-7cf5-45c6-8507-b06d50e6bc13.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyMTczODcsIm5iZiI6MTczOTIxNzA4NywicGF0aCI6Ii84MzY0OTAyMi80MTE0NTk2ODctODk2YzRhZjAtN2NmNS00NWM2LTg1MDctYjA2ZDUwZTZiYzEzLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEwVDE5NTEyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyN2Q2ZDEwN2QxZDZmMjM2MzViNTQ2OWE0OWM3NDIxNmU5NDRjNjNiYTQ3MTJiZDdjN2RmNTM3ODhiOWI0MjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6LSk9VvdWDmEebDV2H-jEzJy32-oY1x3lPXQWO9ByBc)
> 
> I also attach the error in the Developer Tools window after I click on the restart icon.
> 
> ![Image](https://github.com/user-attachments/assets/63717f15-7b5a-4c76-91bb-107e228af796)


I am facing exactly the same issue with Python debugging and it has been driving me insane. Inspecting the console, I also get the same exact DisposableMap issue.

```
lifecycle.ts:790 Error: Trying to add a disposable to a DisposableMap that has already been disposed of. The added object will be leaked!
```

System details:

```
Version: 1.97.0
Commit: 33fc5a94a3f99ebe7087e8fe79fbe1d37a251016
Date: 2025-02-04T22:41:26.688Z
Electron: 32.2.7
ElectronBuildId: 10660205
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Darwin arm64 24.2.0
```

--- a/src/vs/workbench/contrib/debug/browser/repl.ts
+++ b/src/vs/workbench/contrib/debug/browser/repl.ts
@@ -215,6 +215,8 @@ export class Repl extends FilterViewPane implements IHistoryNavigationWidget {
 			const focusedSession = this.debugService.getViewModel().focusedSession;
 			if (this.tree && this.tree.getInput() !== focusedSession) {
 				this.onDidFocusSession(focusedSession);
+			} else {
+				this.selectSession();
 			}
 
 			this.setMode();

